src/jit/morph.cpp

```diff
@@ -6628,9 +6628,10 @@ GenTreePtr Compiler::fgMorphField(GenTreePtr tree, MorphAddrContext* mac)
                 else
 #endif // _TARGET_64BIT_
                 {
-                    // Only volatile could be set, and it maps over
-                    noway_assert((tree->gtFlags & ~(GTF_FLD_VOLATILE | GTF_COMMON_MASK)) == 0);
-                    noway_assert(GTF_FLD_VOLATILE == GTF_IND_VOLATILE);
+                    // Only volatile or classinit could be set, and they map over
+                    noway_assert((tree->gtFlags & ~(GTF_FLD_VOLATILE | GTF_FLD_INITCLASS | GTF_COMMON_MASK)) == 0);
+                    noway_assert(GTF_FLD_VOLATILE == GTF_CLS_VAR_VOLATILE);
+                    noway_assert(GTF_FLD_INITCLASS == GTF_CLS_VAR_INITCLASS);
```

Comment:
I see that this is repeating an existing pattern, but it seems weird to have a `noway_assert` for a condition that is statically checkable, though I guess a `#if` and `#error` combination would probably be less readable.